applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/server_store.c

```diff
@@ -588,26 +524,103 @@ bool srv_store_preset_validated(struct bt_audio_codec_cfg const *const new,
 	return false;
 }
 
+struct foreach_stream_data {
+	int ret;
+	struct bt_bap_ep_info *incoming_ep_info;
+	struct bt_bap_qos_cfg_pref *common_qos;
+	struct bt_bap_qos_cfg_pref const *incoming_qos;
+	struct bt_bap_stream *incoming_stream;
+	uint8_t *streams_checked;
+	uint32_t *existing_pres_dly_us;
+	uint32_t *computed_pres_dly_us;
+	uint32_t *existing_pres_dly_us_check;
+	bool *group_reconfig_needed;
+	bool existing_pres_dly_already_in_range;
+};
+
+static bool foreach_stream_in_group(struct bt_cap_stream *existing_stream, void *user_data)
+{
+	int ret;
+	struct foreach_stream_data *ctx = (struct foreach_stream_data *)user_data;
+
+	if (existing_stream->bap_stream.group != ctx->incoming_stream->group) {
+		/* The existing stream is not in the same group as the incoming stream */
+		LOG_ERR("Existing stream not in same group as incoming stream");
+		ctx->ret = -EINVAL;
+		return true;
+	}
+
+	if (pres_dly_stream_ignore(&(existing_stream->bap_stream), ctx->incoming_stream)) {
+		/* Do not consider this stream. Continue parsing */
+		LOG_WRN("Ignoring existing stream");
+		ctx->ret = 0;
+		return false;
+	}
+
+	/* All already running streams in the same direction and in the
+	 * same group shall have the same presentation delay.
+	 */
+	*ctx->existing_pres_dly_us = existing_stream->bap_stream.qos->pd;
+
+	if (*ctx->existing_pres_dly_us_check == UINT32_MAX) {
+		ctx->existing_pres_dly_us_check = ctx->existing_pres_dly_us;
+	} else if (*ctx->existing_pres_dly_us_check != *ctx->existing_pres_dly_us) {
+		LOG_ERR("Illegal value. Pres delays do not match: %u != %u",
+			*ctx->existing_pres_dly_us_check, *ctx->existing_pres_dly_us);
+		ctx->ret = -EINVAL;
+		return true;
+	}
+
+	if (*ctx->existing_pres_dly_us == 0) {
+		LOG_ERR("Existing presentation delay is zero");
+		ctx->ret = -EINVAL;
+		return true;
+	}
+
+	if (IN_RANGE(*ctx->existing_pres_dly_us, ctx->incoming_qos->pd_min,
+		     ctx->incoming_qos->pd_max)) {
+		*ctx->computed_pres_dly_us = *ctx->existing_pres_dly_us;
+		LOG_INF("Existing pres delay is within the incoming stream QoS range");
+		ctx->existing_pres_dly_already_in_range = true;
+		ctx->ret = 0;
+		return true;
+	}
+
+	*ctx->group_reconfig_needed = true;
+	(*ctx->streams_checked)++;
+
+	ret = pres_delay_compute(ctx->common_qos, &existing_stream->bap_stream.ep->qos_pref);
+	if (ret) {
+		ctx->ret = ret;
+		return true;
+	}
```

Comment:
```suggestion
	ctx->ret = pres_delay_compute(ctx->common_qos, &existing_stream->bap_stream.ep->qos_pref);
	if (ctx->ret) {
		return true;
	}
```
Then you can remove this internal `int ret`